third_party/move/move-vm/runtime/src/loader/modules.rs

```diff
@@ -254,17 +261,38 @@ pub(crate) struct StructInstantiation {
     pub(crate) instantiation: Vec<Type>,
 }
 
+#[derive(Clone, Debug)]
+pub(crate) struct StructVariantInfo {
+    // struct field count
+    pub(crate) field_count: u16,
+    pub(crate) variant: VariantCount,
+    pub(crate) definition_struct_type: Arc<StructType>,
+    pub(crate) instantiation: Vec<Type>,
+}
+
 // A field handle. The offset is the only used information when operating on a field
 #[derive(Clone, Debug)]
 pub(crate) struct FieldHandle {
     pub(crate) offset: usize,
+    pub(crate) ty: Type,
     pub(crate) definition_struct_type: Arc<StructType>,
 }
 
 // A field instantiation. The offset is the only used information when operating on a field
 #[derive(Clone, Debug)]
 pub(crate) struct FieldInstantiation {
     pub(crate) offset: usize,
+    pub(crate) uninstantiated_ty: Type,
+    pub(crate) definition_struct_type: Arc<StructType>,
```

Comment:
Would be nice to have consistency between `type` and `ty`. The latter seems better to me

Reply:
Done